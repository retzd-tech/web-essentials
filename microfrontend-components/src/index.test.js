import '@testing-library/jest-dom'
import * as React from 'react'
import { render, fireEvent, screen } from '@testing-library/react'
import renderer from 'react-test-renderer'

import HiddenMessage from '.'

describe('HiddenMessage', () => {
  const testMessage = 'Test Message'

  describe('#render', () => {
    it('should render as expected as snapshot', () => {
      const HiddenMessageSnapshot = renderer
        .create(<HiddenMessage>{testMessage}</HiddenMessage>)
        .toJSON()

      expect(HiddenMessageSnapshot).toMatchSnapshot()
    })
  })

  describe('#click', () => {
    it('should show no message at first', () => {
      render(<HiddenMessage>{testMessage}</HiddenMessage>)

      expect(screen.queryByText(testMessage)).toBeNull()
    })

    it('should show the hidden message when the checklist input is selected', () => {
      render(<HiddenMessage>{testMessage}</HiddenMessage>)

      expect(screen.queryByText(testMessage)).toBeNull()
      fireEvent.click(screen.getByLabelText(/show/i))
      expect(screen.getByText(testMessage)).toBeInTheDocument()
    })
  })
})
