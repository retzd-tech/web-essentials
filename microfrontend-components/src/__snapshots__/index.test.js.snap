// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HiddenMessage #render should render as expected as snapshot 1`] = `
<div>
  <label
    htmlFor="toggle"
  >
    Show Message
  </label>
  <input
    checked={false}
    id="toggle"
    onChange={[Function]}
    type="checkbox"
  />
</div>
`;
